pipeline-adapters/mlrun-pipelines-kfp-v1-8/src/mlrun_pipelines/mounts.py

```diff
@@ -39,29 +43,25 @@ def v3io_cred(api="", user="", access_key=""):
         train.apply(use_v3io_cred())
     """
 
-    def _use_v3io_cred(container_op: kfp.dsl.ContainerOp):
+    def _use_v3io_cred(runtime: "KubeResource"):
         from os import environ
 
-        from kubernetes import client as k8s_client
-
         web_api = api or environ.get("V3IO_API") or mlconf.v3io_api
         _user = user or environ.get("V3IO_USERNAME")
         _access_key = access_key or environ.get("V3IO_ACCESS_KEY")
         v3io_framesd = mlconf.v3io_framesd or environ.get("V3IO_FRAMESD")
 
-        return (
-            container_op.container.add_env_variable(
-                k8s_client.V1EnvVar(name="V3IO_API", value=web_api)
-            )
-            .add_env_variable(k8s_client.V1EnvVar(name="V3IO_USERNAME", value=_user))
-            .add_env_variable(
-                k8s_client.V1EnvVar(name="V3IO_ACCESS_KEY", value=_access_key)
-            )
-            .add_env_variable(
-                k8s_client.V1EnvVar(name="V3IO_FRAMESD", value=v3io_framesd)
-            )
+        runtime.spec.env.extend(
+            [
+                {"name": "V3IO_API", "value": web_api},
+                {"name": "V3IO_USERNAME", "value": _user},
+                {"name": "V3IO_ACCESS_KEY", "value": _access_key},
```

Comment:
This potentially exposes the access key on the runtime spec. I don't think it's a good idea to add it as plain text. I'm not sure why not keep this enrichment on the kfp object?

Reply:
I think this will only be exposed if running locally, when the workflow is submitted to the MLRun API, the secret is replaced with a reference to a K8S secret.

Reply:
Right but my point is that after you compile the workflow, the runtime now has the access key as plaintext in its spec on the client side. We should be OK because of the implementation of `mlrun.runtimes.pod.KubeResourceSpec._enrich_field` but IMO it's not good practice to allow this

Reply:
This behavior has not changed, previously, we we would create a ContainerOp, fill it with all the relevant env vars, volumes and volumemounts, and then copy a serialised version (through k8s client) back into the runtime.